client/src/partials/cash/modals/transfer.modal.js

```diff
@@ -0,0 +1,113 @@
+angular.module('bhima.controllers')
+.controller('CashTransferModalController', CashTransferModalController);
+
+CashTransferModalController.$inject = [
+  '$uibModalInstance', '$q', 'SessionService', 'CurrencyService', 'VoucherService', 'CashboxService', 'AccountService', 'cashBox', 'uuid'
+];
+
+/**
+ * @module cash/modals/CashTransferModalController
+ *
+ * @description This controller is responsible transfering money between auxillary cash and a virement account
+*/
+
+function CashTransferModalController(ModalInstance, $q, sessionService, currencyService, voucherService, cashBoxService, accountService, cashBox, uuid) {
+  var vm = this; 
+
+  /** Attaching service to the scope **/
+  vm.cashBox = cashBox;
+  vm.currencyService = currencyService; 
+  vm.cashBoxService = cashBoxService;
+
+  /** check if the cashbox id is provided**/
+  vm.cashBoxIdMissed = !cashBox.id;
+
+  /**
+  * @function submit
+  * @param {boolean} invalid it is true if the form is valid
+  * @description for submitting a dialog content
+  **/
+  function submit (invalid){
+    if (invalid) { return; }
+    var voucher = processVoucher();
+    var voucher_item = processVoucherItem(voucher.uuid);
+    var record = { voucher : voucher, voucher_item : voucher_item };
+    voucherService.create(record)      
+    .then(function (res){
+      vm.submited = true; //FIX ME, I need a goo idea
+    })
+    .catch(function (err){
+      vm.fail = true; //FIX ME, I need a goo idea
+    });          
+  }
+
+  function cancel (){
+    ModalInstance.dismiss();
+  }
+
+  /**
+  * This methode is responsible to generate a description for the voucher operation
+  * @private
+  **/
+  function generateDescription (){
+    return ['Transfer voucher', new Date().toISOString().slice(0, 10)].join('/');
```

Comment:
Why already have the date in a separate column, it might be more useful to store other information.

Shouldn't the person making the transfer be given the option to write their own description?